### PR TITLE
[wasm] Add a function to return the interp trampoline instead of invoking it. Not used yet.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
The interpreter needs to be modified to create some kind of preprocessed data structure to invoke functions of a given signature, this is the first step towards that.

File: mono/mini/aot-runtime-wasm.c [hunk 1]
```diff
@@ -90,6 +90,26 @@ get_long_arg (InterpMethodArguments *margs, int idx)
...
 #include "wasm_m2n_invoke.g.h"
 
+gpointer
+mono_wasm_get_interp_to_native_trampoline (MonoMethodSignature *sig)
+{
+	char cookie [32];
+	int c_count;
+
+	c_count = sig->param_count + sig->hasthis + 1;
+	g_assert (c_count < sizeof (cookie)); //ensure we don't overflow the local
```
Comment (jaykrell) [inline mono/mini/aot-runtime-wasm.c]: alloca it instead, or alloca if >32?

**Comment by lewurm** [comment]
do I understand correctly that you want instead of this
```
interp_main_loop
  -> icall_trampoline_dispatch (cookie, target_func, margs)
  -> wasm_invoke_dd (target_func, margs)
```
have something that directly calls the `wasm_invoke`, so
```
interp_main_loop
  -> wasm_invoke_dd (target_func, margs)
```
that is, moving the `icall_trampoline_dispatch` to the transform phase?

**Comment by vargaz** [comment]
Something like that, need to create a per-signature data structure and embed it into the interpreter code stream.

**Comment by lewing** [comment]
@vargaz do you want to keep this open?

**Pull request closed without merging**